include/arch/arm/arch/32/mode/types.h

```diff
@@ -6,6 +6,8 @@
 
 #pragma once
 
+#include <stdint.h>
```

Comment:
Are you aware that all of the C code in the kernel is compiled as a single C file? All files are concatenated together into kernel_all.c and then compiled in one go.  This means that moving these header files around can change the kernel in unexpected ways.  Something we'd be interested in doing though would be having a kernel build option that compiles each file individually.  Once we have this then it'd be a lot easier to reorganise includes without breaking things on certain configurations

Reply:
> Something we'd be interested in doing though would be having a kernel build option that compiles each file individually. 

If we do this, we need to be careful with the verified configurations. We want the compiler to see the same input that the verification toolchain sees.

Reply:
I should have clarified that this wouldn't be for verified configurations.  (Although, would translation validation be able to show that a kernel.elf assembled from many .o is the same as the original source that had been assembled as a single C source file?

Reply:
Translation validation should be able to deal with that, yes (but we only have this for 2 of the verified platforms)